include/itkAttenuationImageFilter.hxx

```diff
@@ -204,6 +246,56 @@ AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::ComputeAttenuatio
   return 10 * neper * frequencySlope / distanceMM; // 10 converts mm into cm
 }
 
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+float
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::TransformPhysicalToPixelScanLineDistance(
+  float distanceMM) const
+{
+  if (distanceMM < 0)
+  {
+    itkExceptionMacro("Expected nonnegative spatial distance!");
+  }
+
+  auto input = this->GetInput();
+  if (input == nullptr)
+  {
+    itkExceptionMacro("Tried to translate spatial distance to pixel distance without reference input image!");
+  }
+
+  const float scanStepMM = input->GetSpacing()[m_Direction];
+  float       distanceInPixels = distanceMM / scanStepMM;
+  return static_cast<unsigned int>(std::round(distanceInPixels));
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+float
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::TransformPixelToPhysicalScanLineDistance(
+  unsigned int distance) const
+{
+  auto input = this->GetInput();
+  if (input == nullptr)
+  {
+    itkExceptionMacro("Tried to translate spatial distance to pixel distance without reference input image!");
+  }
+
+  const float scanStepMM = input->GetSpacing()[m_Direction];
+  return scanStepMM * distance;
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+bool
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::IsIncluded(InputIndexType index) const
+{
+  auto maskImage = this->GetMaskImage();
```

Comment:
This method is called a lot. I suggest caching this as iVar at the start of generate data (in `BeforeThreadedGenerateData`?).

Reply:
Seems reasonable, added.